src/components/App/App.js

```diff
@@ -26,13 +28,31 @@ const App = () => {
           setError(data);
           return;
         }
-        setCurrentActivity(data);
+        setCurrentActivity({...data, done: false});
       });
   }
 
   const addActivity = () => {
     const newActivities = [currentActivity, ...activities];
-    setActivities(newActivities);
+    setLocalActivites(newActivities);
```

Comment:
I like this flow of setting the local storage first and using the function that does that to set your state.